source/extensions/filters/http/squash/squash_filter.cc

```diff
@@ -275,7 +277,8 @@ void SquashFilter::pollForAttachment() {
   request->headers().setReferenceHost(SERVER_AUTHORITY);
 
   in_flight_request_ =
-      cm_.httpAsyncClientForCluster(config_->clusterName())
+      cm_.getThreadLocalCluster(config_->clusterName())
```

Comment:
do we need another thread local cluster check here?

Reply:
Technically I think yes, but I couldn't quickly sort out how to test this and this code is not clearly used/owned so I will admit I punted. Also, it's guarded behind a previous check, so it would be a rare condition to hit this becoming null but it could happen. Let me add a TODO for this to follow up later since it's a pre-existing bug.